### PR TITLE
Update the algorithm that calculate weights for WeightedRandomSampler

### PR DESCRIPTION
**Pull request by tybalex**
What does this PR do:
1. Update the algorithm that calculate weights for WeightedRandomSampler
2. incorporate the changes that remove the unnecessary `is_control_plane_log` parameter in masker.py

### CODE REVIEW

File: models/opnilog/opnilog_model.py [hunk 1]
```diff
@@ -301,28 +299,19 @@ def __init__(
...
-        #                                feature_range=(0.005, 0.995))
-        weights = 1 - minmax_scale(
-            data_token_idx_df.sum(axis=1), feature_range=(0.0, 0.75)
-        )
+        """
+        This function aims to address the case the data is very imbalanced, and make the data less imbalanced.
+        """
+        unique_sample_counter = defaultdict(int)
+        str_data = [str(l) for l in self.data]
+        for s in str_data:
```
Comment (sanjay920) [inline models/opnilog/opnilog_model.py]: this is inefficient. can loop self.data and can str() inside the loop to avoid unnecessary storage for `str_data`

**Review by sanjay920 — CHANGES_REQUESTED** [review]

File: models/opnilog/opnilog_model.py (hunk 1 above)
Comment (tybalex) [reply models/opnilog/opnilog_model.py]: Not really, `str_data` is used again @ line 314.

File: models/opnilog/opnilog_parser.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import time
 
 # Third Party
+import pandas as pd
```
Comment (AmartC) [inline models/opnilog/opnilog_parser.py:7]: Why do we need to import pandas here?

Comment (tybalex) [reply models/opnilog/opnilog_parser.py:7]: the IDE added it automatically as it being used in one of the function in this script.

File: models/opnilog/opnilog_model.py (hunk 1 above)
Comment (sanjay920) [reply models/opnilog/opnilog_model.py]: you can loop `self.data`. saving storage >>> saving compute here
306-308:
```
for s in self.data:
   unique_sample_counter[str(s)] += 1
```
314: `weights = np.array([unique_sample_counter[str(s)] for s in self.data])`

Comment (tybalex) [reply models/opnilog/opnilog_model.py]: saving storage is not >>> compute here bc `self.data` contains numbers of valid tokens not the actual string, so it's much less temporal memory pressure than you think. But if you insist, sure.

Comment (sanjay920) [reply models/opnilog/opnilog_model.py]: okay then why not directly modify type of `self.data` ? you'd get what you want without sacrificing any storage

**Review by sanjay920 — APPROVED** [review]

**Review by AmartC — APPROVED** [review]
LGTM but please wait to merge.

**Comment by tybalex** [comment]
> LGTM but please wait to merge.

why?

**Pull request merged**